snakesist/exist_client.py

```diff
@@ -101,6 +83,77 @@ def node_id(self):
         return self._node_id
 
 
+class DocumentResource(Resource):
+    """
+    A representation of an eXist document node
+    """
+    def __init__(
+            self,
+            exist_client: "ExistClient",
+            query_result: Optional[QueryResultItem] = None
+    ):
+        """
+        :param exist_client: The client to which the resource is coupled.
+        :query_result: A tuple containing the absolute resource ID, node ID
+                       and the node of the resource.
+        """
+        super().__init__(exist_client, query_result)
```

Comment:
The methods on the concrete classes aren't necessary when they only call the superclass' `__init__` w/ unaltered arguments. however, they could contain checks for constraints of the result that gets processed, e.g. this class expects no `node_id`.

Reply:
i refactored that redundancy away and inserted a check for `DocumentResource`. in the meantime i found out that document root nodes in eXist always have a `node_id` of `"1"`, so the check throws a `ValueError` if the value is not `"1"`. but i assume the `node_id` could be dropped completely for `DocumentResource`...